File: Exercicio 1/exercise_1.py
alunos = []

print("Preencha com as informações de cada aluno, para parar a execução escreva 'exit' ao preencher o nome!")
print("")
while True:
    aluno = input(f"Digite o nome do aluno N°{len(alunos) + 1}: ")

    if aluno == "exit":
        break

    idade = input(f"Digite a idade do aluno N°{len(alunos) + 1}: ")
    nota1 = input(f"Digite a nota do 1º bimestre do aluno N°{len(alunos) + 1}: ")
    nota2 = input(f"Digite a nota do 2º bimestre do aluno N°{len(alunos) + 1}: ")
    nota3 = input(f"Digite a nota do 3º bimestre do aluno N°{len(alunos) + 1}: ")
    nota4 = input(f"Digite a nota do 4º bimestre do aluno N°{len(alunos) + 1}: ")

    alunos.append([aluno,
                   int(idade),
                   [int(nota1), int(nota2), int(nota3), int(nota4)]])

print("")
print("===========================================")
print(f"Média de idade de todos os alunos:")
idadeTotal = 0
for aluno, idade, notas in alunos:
    idadeTotal = idadeTotal + idade
print(idadeTotal/len(alunos))

print("")
print(f"Média de nota de todos os alunos:")
for aluno, idade, notas in alunos:
    print(f"- {aluno}: {sum(notas)/4}")

print("")
print("===========================================")
print("Situação final de cada aluno:")
for aluno, idade, notas in alunos:
    situacao = "Reprovado" if sum(notas)/4 < 4 else ("Recuperação" if sum(notas)/4 < 7 else "Aprovado")

    print(f"- {aluno}: {situacao}")

File: Exercicio 1/exercise_1.lua
answer = ""
num = 0
lista = {}
totalIdades = 0
-------------------------------------------------------
-- LEITURA DAS INFORMAÇÕES
-------------------------------------------------------
while(answer ~= "N") do
  io.write("Escreva o nome de um aluno: ")
  nomeAluno = io.read("*l")
  io.close()
  
  io.write("Escreva a idade do aluno ",nomeAluno,": ")
  idadeAluno = tonumber(io.read("*l"))
  totalIdades = totalIdades + idadeAluno
  io.close()
  print("")

  bi = 1
  notaTotalAluno = 0
  while(bi <= 4) do
    io.write("Qual foi a nota do aluno no ",bi,"º bimestre? ")
    notaTotalAluno = (tonumber(io.read("*l"))*10) + notaTotalAluno

    bi = bi + 1
  end
  print("")

  table.insert(lista,{nome=nomeAluno;idade=idadeAluno;media=notaTotalAluno/4})

  num = num + 1

  io.write("Deseja inserir mais algum aluno? (S/N) ")
  answer = io.read("*l")
  io.close()
end
---------------------------------------------------------
-- CÁLCULO E IMPRESSÃO DA MÉDIA DA IDADE DOS ALUNOS
---------------------------------------------------------
print("")
print("========================")
print("||       RESUMO       ||")
print("========================")
print("")

rounded_age = math.ceil((totalIdades*10)/num)/10
print("Média das idades dos alunos: "..rounded_age)
print("")
---------------------------------------------------------
-- CÁLCULO DA MÉDIA E IMPRESSÃO DAS INFORMAÇÕES DOS ALUNOS
-------------------------------------------------------
for k, v in pairs(lista) do
  print("Nome: "..v.nome,"Idade: "..v.idade)
  rounded_media = math.ceil(v.media)/10
  
  if(rounded_media > 6) then
    print("Média: "..rounded_media,"Situação: Aprovado")
  elseif(rounded_media > 4) then
    print("Média: "..rounded_media,"Situação: Recuperação")
  else
    print("Média: "..rounded_media,"Situação: Reprovado")
  end

  print("")
end
